core/src/main/java/org/hibernate/ogm/hibernatecore/impl/OgmSessionFactoryImpl.java

```diff
@@ -83,4 +93,75 @@ public Reference getReference() throws NamingException {
 				null
 				);
 	}
+
+	@Override
+	public EntityManager createEntityManager() {
+		return new OgmSessionImpl( this, (EventSource) getDelegate().createEntityManager() );
+	}
+
+	@SuppressWarnings("rawtypes")
+	@Override
+	public EntityManager createEntityManager(Map map) {
+		return new OgmSessionImpl( this, (EventSource) getDelegate().createEntityManager( map ) );
+	}
+
+	@Override
+	public EntityManager createEntityManager(SynchronizationType synchronizationType) {
+		return new OgmSessionImpl( this, (EventSource) getDelegate().createEntityManager( synchronizationType ) );
+	}
+
+	@SuppressWarnings("rawtypes")
+	@Override
+	public EntityManager createEntityManager(SynchronizationType synchronizationType, Map map) {
+		return new OgmSessionImpl( this, (EventSource) getDelegate().createEntityManager( synchronizationType, map ) );
+	}
+
+	@Override
+	public <T> void addNamedEntityGraph(String graphName, EntityGraph<T> entityGraph) {
+		throw new IllegalStateException( "Hibernate OGM does not support entity graphs" );
+	}
+
+	@SuppressWarnings("rawtypes")
+	@Override
+	public EntityGraph findEntityGraphByName(String name) {
+		throw new IllegalStateException( "Hibernate OGM does not support entity graphs" );
```

Comment:
No need to change this but I'm surprised you used `IllegalStateException` instead of `UnsupportedOperationException`.

Reply:
IIRC, there were preexisting methods using exactly that so I chose to be consistent.